experimental/dds/tree2/src/shared-tree-core/editManager.ts

```diff
@@ -58,13 +58,16 @@ export class EditManager<
 	 */
 	private readonly trunkMetadata = new Map<
 		RevisionTag,
-		{ sequenceNumber: SeqNumber; sessionId: SessionId }
+		{ sequenceNumber: SeqNumber; clientSequenceNumber: SeqNumber; sessionId: SessionId }
 	>();
 	/**
 	 * A map from a sequence number to the commit in the {@link trunk} which has that sequence number.
 	 * This also includes an entry for the {@link trunkBase} which always has the lowest key in the map.
 	 */
-	private readonly sequenceMap = new BTree<SeqNumber, GraphCommit<TChangeset>>();
+	private readonly sequenceMap = new BTree<
+		SeqNumber,
+		BTree<SeqNumber, GraphCommit<TChangeset>>
```

Comment:
Why did you decide to do nested BTrees rather than uniform keys with a primary and secondary sort?